denops/@denops-private/plugin.ts

```
@@ -0,0 +1,183 @@
+import { join } from "jsr:@std/path@^1.0.2/join";
+import { dirname } from "jsr:@std/path@^1.0.2/dirname";
+
+type PluginModule = {
+  main: Entrypoint;
+};
+
+export class Plugin {
+  #denops: Denops;
+  #loadedWaiter: Promise<void>;
+  #unloadedWaiter?: Promise<void>;
+  #disposable: AsyncDisposable = voidAsyncDisposable;
```

Biome (1.9.4)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)